Issue 3624 blog v0.23 (#3725)
knative/docs: blog/releases/announcing-knative-v0-23-release.md

---

> title: "Version v0.23 release"

Reviewer: I think the idea was that it's just v0.2x rather than "Version v0.2x" 🙂 sorry. If you don't like this as much as "Version 0.2x" I'd raise it with @omerbensaadon maybe since it was added to the word list that way. I don't care which we use as long as it's consistent.

Author: No problem, I will make the update.
I will go back and remove the "Version" from all.

Author: @abrennan89 I remove "Version " from all titles and linktitles so all are consistent now

> linkTitle: "Version 0.23 release"

Reviewer: see above and https://docs.google.com/spreadsheets/d/1p1_kBUd6ZvonxHkMcEJPayf6QIpExuFf5cFq0ptar7I/edit#gid=0

type: "blog"
image: knative-eventing.png
---

### Announcing Knative v0.23 Release

A new version of Knative is now available across multiple components.
Follow the instructions in the documentation [Installing Knative](https://knative.dev/docs/install/) for the respective component.

#### Table of Contents
- [Serving v0.23](#serving-v023)
- [Eventing v0.23](#eventing-v023)
- Eventing Extensions
    - [Apache Kafka Broker v0.23](#apache-kafka-broker-v023)
    - [RabbitMQ Eventing v0.23](#rabbitmq-eventing-v023)
- `kn` [CLI v0.23](#client-v023)
- [Operator v0.23](#operator-v023)
- [Thank you contributors](#thank-you-contributors)

### Highlights

- Serving improves support for Istio when using Mesh to access the user container directly
- Eventing removes old API version only leaving `v1` as the one available
- The `kn` CLI and Operator keeps crushing bugs in this release.

### Serving v0.23

<!-- Original notes are here: https://github.com/knative/serving/releases/tag/v0.23.0 -->

#### 🚨 Breaking or Notable Changes

- Change the default post-install job to use sslip.io rather than xip.io. [#11298](https://github.com/knative/serving/pull/11298)

#### 💫 New Features & Changes

> - The stats scraping in the autoscaler is now sensitive to the EnableMeshPodAddressability setting. A restart of the autoscaler is required for the setting to take effect if changed. [#11161](https://github.com/knative/serving/pull/11161)
> - The state keeping in the activator is now sensitive to the EnableMeshPodAddressability setting. A restart of the activator is required for the setting to take effect if changed. [#11172](https://github.com/knative/serving/pull/11172)
> - Tightens the heuristic for mesh being enabled in the service scraper. We now expect all errors to be related to mesh (i.e. 503 status code). This prevents accidentally falling in to service scrape mode when errors are encountered for other reasons. [#11174](https://github.com/knative/serving/pull/11174)

Reviewer: ```suggestion
- The stats scraping in the autoscaler is now sensitive to the `EnableMeshPodAddressability` setting. A restart of the autoscaler is required for the setting to take effect if changed. [#11161](https://github.com/knative/serving/pull/11161)
- The state keeping in the activator is now sensitive to the `EnableMeshPodAddressability` setting. A restart of the activator is required for the setting to take effect if changed. [#11172](https://github.com/knative/serving/pull/11172)
- Tightens the heuristic for mesh being enabled in the service scraper. We now expect all errors to be related to mesh (i.e. 503 status code). This prevents accidentally falling in to service scrape mode when errors are encountered for other reasons. [#11174](https://github.com/knative/serving/pull/11174)
```

#### 🐞 Bug Fixes

- Added schemas to all CRDs. [#11244](https://github.com/knative/serving/pull/11244)
- Changed the rollout behavior of application deployment changes (due to Knative upgrade for example) to never have less ready posd than required. [#11140](https://github.com/knative/serving/pull/11140)
- Rate limits digest resolution (10 QPS, retry back-off 1s to 1000s) to prevent exceeding quota at remote registries [#11279](https://github.com/knative/serving/pull/11279)

> - Revision replicas shut down 15s quicker. [#11249](https://github.com/knative/serving/pull/11249)

Reviewer: ```suggestion
- Revision replicas now shut down 15 seconds faster. [#11249](https://github.com/knative/serving/pull/11249)
```

> - The activator's proxy is now sensitive to the EnableMeshPodAddressability setting. [#11162](https://github.com/knative/serving/pull/11162)
> - Update the User-Agent used during tag resolution [#10590](https://github.com/knative/serving/pull/10590)

Reviewer: ```suggestion
- The activator proxy now recognizes the `EnableMeshPodAddressability` setting. [#11162](https://github.com/knative/serving/pull/11162)
- Update the User-Agent used during tag resolution [#10590](https://github.com/knative/serving/pull/10590)
```

### Eventing v0.23

<!-- Original notes are here: https://github.com/knative/eventing/releases/tag/v0.23.1 -->

#### 🚨 Breaking or Notable Changes

- Remove {`eventing`,`flows`,`messaging`} `v1beta1` APIs [#5201](https://github.com/knative/eventing/pull/5201)
- Promote PingSource to v1 [#5324](https://github.com/knative/eventing/pull/5324)
- Remove `APIServerSource`, `ContainerSource`, `PingSource` and `SinkBinding` v1alpha2 APIs [#5318](https://github.com/knative/eventing/pull/5318)
- Remove `APIServerSource`, `ContainerSource`, `PingSource` and `SinkBinding` v1beta1 APIs [#5319](https://github.com/knative/eventing/pull/5319)

> - Remove `APIServerSource` and `SinkBinding` v1alpha1 APIs [#5317](https://github.com/knative/eventing/pull/5317)

Reviewer: Is there any docs work to do here to make sure any sample YAMLs are showing the right API version?

Author: Don't know @knative/eventing-wg-leads would need to look into that

Reviewer: Ville checked this and put in a PR

#### 💫 New Features & Changes

> - `InMemoryChannel` can be used independent of Eventing Core. Install by using `in-memory-channel.yaml` no need to install anything else to be able to use the feature. [#5355](https://github.com/knative/eventing/pull/5355)

Reviewer: ```suggestion
- InMemoryChannel can now be used independently of Knative Eventing, and can be installed by applying the `in-memory-channel.yaml`. [#5355](https://github.com/knative/eventing/pull/5355)
```

Reviewer: Should there be docs for things like this?

Author: Yes they should write docs for this like this @knative/eventing-wg-leads

> - `MTChannelBasedBrokers` will see improved filter performance for triggers with few or no trigger filters [#5288](https://github.com/knative/eventing/pull/5355)

Reviewer: ```suggestion
- Multi-tenant, channel-based brokers now have improved filter performance for Triggers with few or no filters. [#5288](https://github.com/knative/eventing/pull/5355)
```

How many is "few", can we specify? e.g. "up to X filters"?

Author: Don't know, but @n3wscott  might know

#### 🐞 Bug Fixes

- `duckv1.SourceSpec`, `TimeZone`, `Schedule` will be populated when called by `v1beta1/v1alpha2` clients [#5153](https://github.com/knative/eventing/pull/5153)

> - Propagate `trigger.spec.delivery` to subscriptions for `MTChannelBasedBroker`. [#5267](https://github.com/knative/eventing/pull/5267)

Reviewer: ```suggestion
- The Trigger `delivery` spec is now automatically propagated to Subscriptions that are connected to multi-tenant, channel-based Brokers. [#5267](https://github.com/knative/eventing/pull/5267)
```

Reviewer: idk if some of these are accurate but I'm adding suggestions to improve the wording if this is what it means

Author: yes this reads better 👍

Author: And your input is helping a lot @abrennan89

#### 🧹 Clean up

> - Tools from https://github.com/knative/eventing-contrib/tree/main/cmd were moved to eventing repo. [#5295](https://github.com/knative/eventing/pull/5295)

Reviewer: ```suggestion
- Tools from the [`eventing-contrib` repository](https://github.com/knative/eventing-contrib/tree/main/cmd) have been moved to the `eventing` repository. [#5295](https://github.com/knative/eventing/pull/5295)
```

> - Sequence now displays conditions as Unknown rather than False for non-terminal condition states. [#5369](https://github.com/knative/eventing/pull/5369)

Reviewer: ```suggestion
- Sequence now displays conditions as `Unknown` rather than `False` for non-terminal condition states. [#5369](https://github.com/knative/eventing/pull/5369)
```

> - Running validation for event ingress on mt broker [#5275](https://github.com/knative/eventing/pull/5275)

Reviewer: ```suggestion
- Running validation for event ingress on multi-tenant broker [#5275](https://github.com/knative/eventing/pull/5275)
```

- Use fully qualified test images. [#1307](https://github.com/knative/client/pull/1307)
- Fix documentation of the configuration options. [#1297](https://github.com/knative/client/pull/1297)
- Respect -o in list commands in case if no data present. [#1276](https://github.com/knative/client/pull/1276)

### Operator v0.23

<!-- Original notes are here: https://github.com/knative/operator/releases/tag/v0.23.0 -->

#### 🐞 Bug Fixes

- Apply high availability settings to kourier as well. [#579](https://github.com/knative/operator/pull/579)
- Clear OwnerReferences on knative-local-gateway Service. [#567](https://github.com/knative/operator/pull/567)
- Set the namespace of the service knative-local-gateway to the istio namespace. [#590](https://github.com/knative/operator/pull/590)
- Add pingsource-mt-adapter into HA list. [#591](https://github.com/knative/operator/pull/591)
- Return an actual newest version of the directory latest is not available. [#576](https://github.com/knative/operator/pull/576)
- Allow for all Ingresses to be disabled. [#571](https://github.com/knative/operator/pull/571)

#### 🧹 Clean up

> - Simplify istio installation for e2e test. [#564](https://github.com/knative/operator/pull/564)

Reviewer: ```suggestion
- Simplify Istio installation for end-to-end test. [#564](https://github.com/knative/operator/pull/564)
```

> - Use deployment namespace to determine gateway namespace for Kourier. [#577](https://github.com/knative/operator/pull/577)

Reviewer: ```suggestion
- Use the `deployment` namespace to determine gateway namespace for Kourier. [#577](https://github.com/knative/operator/pull/577)
```
Is it actually called `deployment`?

Author: Don't think is a namespace named `deployment`
Looking at the code in the [PR 577](https://github.com/knative/operator/pull/577/files#diff-b4c139ccd3f76f7257855c955ca2d3bceaf5786788d965f88fbcfc34a2d3585dR46)  is setting an environment variable to the namespace value where the kourier gateway deployment is deploy.

> - Update the script of the upgrade tests for serving. [#574](https://github.com/knative/operator/pull/574)

Reviewer: ```suggestion
- Updated the Knative Serving upgrade tests script. [#574](https://github.com/knative/operator/pull/574)
```
